### PR TITLE
controller,cli: Update route

### LINKED ISSUE #981: controller: Client has no method to update an existing route
The controller has `RouteList`, `GetRoute`, `CreateRoute`, and `DeleteRoute`, but no `UpdateRoute`. It should probably be added (but maybe after #959 is merged :)

### PR DESCRIPTION
**Pull request by jvatic**
Closes #981

### CODE REVIEW

File: cli/route.go [hunk 1]
```diff
@@ -162,6 +176,81 @@ func runRouteAddHTTP(args *docopt.Args, client *controller.Client) error {
...
+	appName := mustApp()
+
+	route, err := client.GetRoute(appName, id)
+	if err != nil {
+		return err
+	}
+
+	if service := args.String["--service"]; service != "" {
+		route.Service = service
+	}
```
Comment (titanous) [inline cli/route.go]: This should probably exit early if the service is the same or empty.

File: cli/route.go [hunk 2]
```diff
@@ -162,6 +176,81 @@ func runRouteAddHTTP(args *docopt.Args, client *controller.Client) error {
...
+
+	tlsCertPath := args.String["--tls-cert"]
+	tlsKeyPath := args.String["--tls-key"]
+	if tlsCertPath != "" && tlsKeyPath != "" {
```
Comment (titanous) [inline cli/route.go]: This logic looks similar to add, can it be abstracted?

**Comment by jvatic** [comment]
@titanous comments addressed

**Comment by titanous** [comment]
Do we have any integration tests for configuring HTTPS routes with the CLI? If not, please add them since that functionality is important.

**Comment by jvatic** [comment]
@titanous Integration test added

**Comment by titanous** [comment]
LGTM

**Pull request merged**